File: emip_dataset/EMIP_DataCollection_Materials/emip_stimulus_programs/Vehicle.scala
class Vehicle (p: String, t: String, tp: Int){
  
  val producer = p
  val vtype = t
  val topSpeed = tp
  var currentSpeed = 0
  
  def accelerate(kmh: Int) = {
    if ((this.currentSpeed + kmh) > this.topSpeed) {
      this.currentSpeed = this.topSpeed
    } else {
      this.currentSpeed = this.currentSpeed + kmh
    }    
    this.currentSpeed
  }
  
}

object VehicleApp extends App {
  val v = new Vehicle("Audi", "A6", 200)
  v.accelerate(10)
}

File: emip_dataset/EMIP_DataCollection_Materials/emip_stimulus_programs/Vehicle.java
public class Vehicle {

	String producer , type ;

	int topSpeed , currentSpeed ;


	public Vehicle ( String p , String t , int tp ) {

		this.producer = p ;

		this.type = t ;

		this.topSpeed = tp ;

		this.currentSpeed = 0 ;

	}


	public int accelerate ( int kmh ) {

		if ( ( this.currentSpeed + kmh ) > this.topSpeed ) {

			this.currentSpeed = this.topSpeed ;

		} else {
	
			this.currentSpeed = this.currentSpeed + kmh ;

		}

		return this.currentSpeed ;

	}


	public static void main ( String args [ ] ) {

		Vehicle v = new Vehicle ( "Audi" , "A6" , 200 ) ;

		v.accelerate ( 10 ) ;

	}

}
